### DISCUSSION TITLE
openlit through grpc port 4317

### DISCUSSION THREAD

**Discussion by 287611** [openlit/openlit#506] [Q&A] [answered]
I am able to get metrics when I connect opentelemetry collector through port 4318. By any chance, does openlit support connecting through port 4317 which is default grpc port

**Comment by amanagarwal042** [comment]
Hey @287611 

Right now it doesnt, As you can see in this line we import the HTTP (4318) exporter https://github.com/openlit/openlit/blob/b4594ff2e83e0e5955b2a712ad70d13fe34f930e/sdk/python/src/openlit/otel/tracing.py#L13

I think adding grpc shouldnt be much different, I can add that support soon in the SDK (if you want to make a PR instead lemme know, Should be a pretty easy addition based on an if condition)

**Comment by amanagarwal042** [comment]
Just raised a PR - https://github.com/openlit/openlit/pull/508

**Reply by 287611** [reply]
Thanks @patcher9 for doing the change

**Comment by amanagarwal042** [comment, marked as answer]
Released [1.30.2](https://pypi.org/project/openlit/1.30.2/) of the Python SDK. 

Now you just need to set `OTEL_EXPORTER_OTLP_PROTOCOL=grpc` in the Environment variables and use the gRPC port/endpoint when passing the OpenTelemetry endpoint.

**Reply by 287611** [reply]
Sure, thanks. Will try the through grpc endpoint

**Reply by 287611** [reply]
I tried running through grpc and it is working now

**Reply by amanagarwal042** [reply]
Super, Thanks for letting me know!
